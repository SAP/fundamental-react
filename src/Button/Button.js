import React from 'react';
import PropTypes from 'prop-types';

export const Button = props => {
  const {
    option,
    type,
    compact,
    glyph,
    dropdown,
    navbar,
    selected,
    disabled,
    typeAttr,
    onclick,
    children
  } = props;
  return (
    <button
      className={`${option ? 'fd-button--' + option : ' fd-button'}${
        type ? ' fd-button--' + type : ''
      }${dropdown ? ' fd-dropdown__control' : ''}${
        compact ? ' fd-button--compact' : ''
      }${glyph ? ' sap-icon--' + glyph : ''}${
        navbar ? ' fd-global-nav__btn' : ''
      }${selected ? ' is-selected' : ''}${disabled ? ' is-disabled' : ''}`}
      selected={selected ? selected : false}
      disabled={disabled ? disabled : false}
      type={typeAttr}
      onClick={onclick}
    >
      {children}
    </button>
  );
};

Button.propTypes = {
  option: PropTypes.oneOf(['', 'emphasized' , 'light', 'shell']),
  type: PropTypes.oneOf(['', 'standard' , 'positive', 'negative', 'medium']),
  compact: PropTypes.bool,
  glyph: PropTypes.string,
  navbar: PropTypes.bool,
  dropdown: PropTypes.bool,
  selected: PropTypes.bool,
  disabled: PropTypes.bool,
  typeAttr: PropTypes.string,
  onclick: PropTypes.func
};

export const ButtonGroup = props => {
  const { children } = props;
  return (
    <div className="fd-button-group" role="group" aria-label="Group label">
      {children}
    </div>
  );
};
