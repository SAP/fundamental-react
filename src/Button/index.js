export { default as Button } from './Button.js';
export { default as ButtonGroup } from './ButtonGroup.js';
