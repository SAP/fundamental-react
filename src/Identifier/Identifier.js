import React from 'react';
import PropTypes from 'prop-types';

export const Identifier = props => {
  const { glyph, size, modifier, color, label, backgroundImageUrl, children } = props;
  const styles = {
    backgroundImage: `url(${backgroundImageUrl})`
  }
  return (
    <span
      className={`${size ? 'fd-identifier--' + size : 'fd-identifier'}${
        glyph ? ' sap-icon--' + glyph : ''
      }${modifier ? ' fd-identifier--' + modifier : ''}${
        color ? ' fd-has-background-color-accent-' + color : ''
      }${backgroundImageUrl ? ' fd-identifier--thumbnail' : ''}`}
      style={backgroundImageUrl && styles}
      role={`${!children ? 'presentation' : ''}`}
      aria-label={label}
    >
      {children}
    </span>
  );
};

Identifier.propTypes = {
  glyph: PropTypes.string,
  size: PropTypes.string,
  modifier: PropTypes.oneOf(['', 'circle', 'transparent']),
  color: PropTypes.oneOf([1, 2, 3, 4, 5, 6, 7, 8, 9]),
  label: PropTypes.string,
  backgroundImageUrl: PropTypes.string
};
