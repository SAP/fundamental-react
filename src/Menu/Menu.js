import classnamesBind from 'classnames/bind';
import MenuItem from './_MenuItem';
import MenuList from './_MenuList';
import PropTypes from 'prop-types';
import React from 'react';
import withStyles from '../utils/withStyles';
import styles from 'fundamental-styles/dist/menu.css';

const classnames = classnamesBind.bind(styles);

/** A **Menu** is the listing structure with optional headers to create menus. Commonly used as the
contents when composing “dropdowns”, “contextual menus”,  etc, when paired with the **Popover**
component. */
const Menu = React.forwardRef(({ children, className, cssNamespace, ...props }, ref) => {

    const menuClasses = classnames(
        `${cssNamespace}-menu`,
        className
    );

    return (
        <nav {...props} className={menuClasses}
            ref={ref}>
            {children}
        </nav>
    );
});

Menu.displayName = 'Menu';

Menu.propTypes = {
    /** Node(s) to render within the component */
    children: PropTypes.node,
    /** CSS class(es) to add to the element */
    className: PropTypes.string
};

Menu.Item = MenuItem;
Menu.List = MenuList;

export default withStyles(Menu);
