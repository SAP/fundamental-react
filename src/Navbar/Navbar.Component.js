import React from 'react';
import {} from '../';
import { DocsTile, DocsText, Separator, Header, Description, Import, Properties } from '..';
import {
    Button,
    Popover,
    Menu,
    MenuList,
    MenuItem,
    Identifier,
    Navbar,
    NavbarGroup,
    NavbarActions,
    NavbarElement
} from '../';

export const NavbarComponent = () => {
    const navbarCode = `<Navbar>
    <NavbarGroup alignment="left">
        <NavbarElement type="side-menu">
            <Button option="light" glyph="menu2" navbar />
        </NavbarElement>
        <NavbarElement type="logo" noMargin="left" />
        <NavbarElement type="product-name">Product Name</NavbarElement>
    </NavbarGroup>

    <NavbarGroup launchpad>
            <Popover
                control={
                    <Button option="light">
                        Suite Name
                    </Button>
                }
                noArrow
                body={
                    <Menu>
                        <MenuList>
                            <MenuItem url="/">Option 1</MenuItem>
                            <MenuItem url="/">Option 2</MenuItem>
                            <MenuItem url="/">Option 3</MenuItem>
                            <MenuItem url="/">Option 4</MenuItem>
                        </MenuList>
                    </Menu>
                }
            />
    </NavbarGroup>

    <NavbarGroup alignment="right">
        <NavbarElement type="context-menu">
            <Popover
                control={<Button option="light">Context Switcher</Button>}
                body={
                    <Menu>
                        <MenuList>
                            <MenuItem url="/">Option 1</MenuItem>
                            <MenuItem url="/">Option 2</MenuItem>
                            <MenuItem url="/">Option 3</MenuItem>
                            <MenuItem url="/">Option 4</MenuItem>
                        </MenuList>
                    </Menu>
                }
            />
        </NavbarElement>

        <NavbarActions>
            <Button option="light" glyph="search" navbar />
            <Button option="light" glyph="action-settings" navbar />
            <Button option="light" navbar>
                <Identifier size="s" modifier="circle">
                    WW
                </Identifier>
            </Button>
        </NavbarActions>
    </NavbarGroup>
</Navbar>`;

    return (
        <div>
            <Header>Navigation Bar</Header>
            <Description>
                The navigation bar (Navbar) component offers a similar navigation approach between Hybris applications.
                It allows for branding, application groupings, access to a left navigation, switch between applications,
                contexts and access tooling such as search or a usersâ€™ profile.
            </Description>
            <Import module='Navbar' path='/fundamental-react/src/' />

            <Separator />

            <Properties
                type='Inputs'
                properties={[
                    {
                        name: 'alignment',
                        description:
                            'string - The position of the group in the navigation bar. Oprions include \'left\' and \'right\'. For default (middle) leave empty.'
                    },
                    {
                        name: 'launchpad',
                        description: 'bool - When set to true, renders a navigation group as a launchpad.'
                    },
                    {
                        name: 'noMargin',
                        description:
                            'string - Renders the NavbarElement with no margin on either left or right side. Oprions include \'left\' and \'right\'. For default leave empty.'
                    },
                    {
                        name: 'type',
                        description:
                            'string (required)- The type of the NavbarElement. Oprions include \'search\', \'context-menu\', \'product-name\', \'logo\', and \'side-menu\'. '
                    }
                ]} />

            <Separator />

            <DocsTile>
                <Navbar>
                    <NavbarGroup alignment='left'>
                        <NavbarElement type='side-menu'>
                            <Button option='light' glyph='menu2'
                                navbar />
                        </NavbarElement>
                        <NavbarElement type='logo' noMargin='left' />
                        <NavbarElement type='product-name'>Product Name</NavbarElement>
                    </NavbarGroup>

                    <NavbarGroup launchpad>
                        <Popover
                            control={<Button option='light'>Suite Name</Button>}
                            noArrow
                            body={
                                <Menu>
                                    <MenuList>
                                        <MenuItem url='/'>Option 1</MenuItem>
                                        <MenuItem url='/'>Option 2</MenuItem>
                                        <MenuItem url='/'>Option 3</MenuItem>
                                        <MenuItem url='/'>Option 4</MenuItem>
                                    </MenuList>
                                </Menu>
                            } />
                    </NavbarGroup>

                    <NavbarGroup alignment='right'>
                        <NavbarElement type='context-menu'>
                            <Popover
                                control={<Button option='light'>Context Switcher</Button>}
                                body={
                                    <Menu>
                                        <MenuList>
                                            <MenuItem url='/'>Option 1</MenuItem>
                                            <MenuItem url='/'>Option 2</MenuItem>
                                            <MenuItem url='/'>Option 3</MenuItem>
                                            <MenuItem url='/'>Option 4</MenuItem>
                                        </MenuList>
                                    </Menu>
                                } />
                        </NavbarElement>

                        <NavbarActions>
                            <Button option='light' glyph='search'
                                navbar />
                            <Button option='light' glyph='action-settings'
                                navbar />
                            <Button option='light' navbar>
                                <Identifier size='s' modifier='circle'>
                                    WW
                                </Identifier>
                            </Button>
                        </NavbarActions>
                    </NavbarGroup>
                </Navbar>
            </DocsTile>
            <DocsText>{navbarCode}</DocsText>

            <Separator />
        </div>
    );
};
