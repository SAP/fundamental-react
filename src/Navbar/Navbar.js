import React from 'react';
import PropTypes from 'prop-types';

// ------------------------------------------- Navbar ------------------------------------------
export const Navbar = props => {
    const { children } = props;
    return (
        <nav className="fd-global-nav">
            {children}
        </nav>
    );
};


// ------------------------------------------- Navbar Group------------------------------------------
export const NavbarGroup = props => {
    const { alignment, launchpad, children } = props;
    return (
        <div className={`fd-global-nav__group${alignment ? ' fd-global-nav__group--' + alignment : ''}${launchpad ? ' fd-global-nav__launchpad' : ''}`}>
            {children}
        </div>
    );
};

NavbarGroup.propTypes = {
    alignment: PropTypes.oneOf(['','left', 'right']),
    launchpad: PropTypes.bool
};

NavbarGroup.defaultProps = {
    alignment: '',
    launchpad: false
};


// ------------------------------------------- Navbar Actions------------------------------------------
export const NavbarActions = props => {
    const { children } = props;
    return (
        <div className='fd-global-nav__actions'>
            {children}
        </div>
    );
};

// ------------------------------------------- Navbar Element------------------------------------------
export const NavbarElement = props => {
    const { noMargin, type, children } = props;
    return (
        <div className={`fd-global-nav__${type}${noMargin ? ' fd-has-margin-' + noMargin + '-none' : ''}`}>
            {children}
        </div>
    );
};
NavbarElement.propTypes = {
    type: PropTypes.oneOf(['search', 'context-menu', 'product-name', 'logo', 'side-menu']).isRequired,
    noMargin: PropTypes.oneOf(['','left', 'right'])
};