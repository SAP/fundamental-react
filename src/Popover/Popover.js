import chain from 'chain-function';
import classnamesBind from 'classnames/bind';
import CustomPropTypes from '../utils/CustomPropTypes/CustomPropTypes';
import { findDOMNode } from 'react-dom';
import FocusManager from '../utils/focusManager/focusManager';
import { isFocusable } from 'tabbable';
import keycode from 'keycode';
import Popper from '../utils/_Popper';
import PropTypes from 'prop-types';
import shortId from '../utils/shortId';
import withStyles from '../utils/withStyles';
import { GridSelector, POPOVER_TYPES, POPPER_PLACEMENTS, POPPER_SIZING_TYPES } from '../utils/constants';
import React, { Component } from 'react';
import styles from 'fundamental-styles/dist/popover.css';

const classnames = classnamesBind.bind(styles);

/** A **Popover** is a wrapping component that accepts a "control" as well as a "body". A control can be
anything that you want to trigger the interaction from. The body will be the contents of what you reveal
on the page after triggering the popover. When paired with the **Menu** component, the Popover is commonly
used as the interaction/wrapping component for composing "dropdowns", "contextual menus", etc. As a general rule,
it is suggested that one Popover be revealed on the page at any given time. Opening one Popover should close all
others to prevent multiple layers and collisions of several popovers. */

class Popover extends Component {
    constructor(props) {
        super(props);

        this.state = {
            isExpanded: false
        };

        //A generated shortId as fallback, in case props.popperProps.id is unset.
        //This ID binds the popover and its control by 'aria-controls'.
        this.popoverId = shortId.generate();
    }

    isButton = (node) => {
        if (!node) {
            return false;
        }
        if (typeof node.type === 'string') {
            return node.type.toLowerCase() === 'button';
        } else if (node.type.displayName) {
            return node.type.displayName.toLowerCase() === 'button';
        }
        return false;
    };

    triggerBody = () => {
        if (!this.props.disabled) {
            this.setState(prevState => ({
                isExpanded: !prevState.isExpanded
            }));
        }
    };

    handleFocusManager = () => {
        if (this.state.isExpanded && this.popover && this.controlRef) {
            this.focusManager = new FocusManager(this.popover, this.controlRef, this.props.useArrowKeyNavigation, this.props.firstFocusIndex);
        }
    };

    handleOutsideClick = () => {
        if (this.state.isExpanded) {
            this.setState({
                isExpanded: false
            });
        }
    };

    handleEscapeKey = () => {
        this.handleOutsideClick();

        if (this.controlRef) {
            if (isFocusable(this.controlRef)) {
                this.controlRef.focus();
            } else {
                const firstTabbableNode = this.controlRef.querySelectorAll(GridSelector.FOCUSABLE)[0];
                firstTabbableNode && firstTabbableNode.focus();
            }
        }
    };

    handleKeyPress = (event, node, onClickFunctions) => {
        if (!this.isButton(node)) {
            switch (keycode(event)) {
                case 'enter':
                case 'space':
                    event.preventDefault();
                    onClickFunctions();
                    break;
                default:
            }
        }
    };

    render() {
        const {
            disableEdgeDetection,
            disableKeyPressHandler,
            disableTriggerOnClick,
            flipContainer,
            firstFocusIndex,
            onClickOutside,
            onEscapeKey,
            disabled,
            noArrow,
            control,
            body,
            className,
            cssNamespace,
            placement,
            popperClassName,
            popperProps,
            widthSizingType,
            useArrowKeyNavigation,
            type,
            show,
            modalManager,
            ...rest
        } = this.props;

        let onClickFunctions = control?.props.onClick;
        if (!disableTriggerOnClick) {
            onClickFunctions = this.triggerBody;

            if (control.props.onClick) {
                onClickFunctions = chain(this.triggerBody, control.props.onClick);
            }
        }

        const id = popperProps.id || this.popoverId;

        let newControlProps = {
            onClick: onClickFunctions,
            ref: (c) => {
                this.controlRef = findDOMNode(c);
                if (control.ref) {
                    if (typeof control.ref === 'function') {
                        control.ref(c);
                    } else if (typeof control.ref === 'object') {
                        control.ref.current = c;
                    }
                }

            }
        };

        const innerRef = (c) => {
            this.popover = findDOMNode(c);
            this.handleFocusManager();
        };

        if (!disableKeyPressHandler) {
            newControlProps = {
                ...newControlProps,
                tabIndex: 0,
                role: !!control?.props.role ? control?.props.role : 'button',
                'aria-controls': id,
                'aria-expanded': this.state.isExpanded,
                'aria-haspopup': !!type ? type : true,
                onKeyPress: (event) => this.handleKeyPress(event, control, onClickFunctions)
            };
        }

        const referenceClassName = classnames(`${cssNamespace}-popover__control`, {
            'is-expanded': this.state.isExpanded
        });

        const referenceComponent = React.cloneElement(control, newControlProps);

        const popoverClasses = classnames(`${cssNamespace}-popover`, className);

        return (
            <div {...rest} className={popoverClasses}>
                <Popper
                    cssBlock={`${cssNamespace}-popover`}
                    disableEdgeDetection={disableEdgeDetection}
                    flipContainer={flipContainer}
                    innerRef={innerRef}
                    modalManager={modalManager}
                    noArrow={noArrow}
                    onClickOutside={chain(this.handleOutsideClick, onClickOutside)}
                    onEscapeKey={chain(this.handleEscapeKey, onEscapeKey)}
                    popperClassName={popperClassName}
                    popperPlacement={placement}
                    popperProps={{ ...popperProps, id }}
                    referenceClassName={referenceClassName}
                    referenceComponent={referenceComponent}
                    show={!disabled && (typeof show === 'boolean' ? show : this.state.isExpanded)}
                    usePortal
                    widthSizingType={widthSizingType}>
                    {body}
                </Popper>
            </div>
        );
    }
}

Popover.displayName = 'Popover';

Popover.propTypes = {
    /** Node(s) to render in the overlay */
    body: PropTypes.node.isRequired,
    /** Node to render as the reference element (that the `body` will be placed in relation to) */
    control: PropTypes.node.isRequired,
    /** CSS class(es) to add to the element */
    className: PropTypes.string,
    /** Set to **true** to mark component as disabled and make it non-interactive */
    disabled: PropTypes.bool,
    /** Set to **true** to render popover without edge detection so popover will not flip from top to bottom with scroll */
    disableEdgeDetection: PropTypes.bool,
    /** Set to **true** to remove onKeyPress handler and aria-* roles.
     * Only do so if the control is a complex component such as a FormInput with Button */
    disableKeyPressHandler: PropTypes.bool,
    /** Set to **true** to remove default triggerBody handler used in onClick.
     * Useful for when a custom method is desired to open the Popover */
    disableTriggerOnClick: PropTypes.bool,
    /** Index of the focusable item to focus first within the Popover */
    firstFocusIndex: PropTypes.number,
    /** The bounding container to use when determining if the popover is out of bounds */
    flipContainer: CustomPropTypes.elementOrArrayOfElements(),
    /** If Popover is to be rendered in a modal, the parent modal manager can be passed as a prop */
    modalManager: PropTypes.object,
    /** Set to **true** to render a popover without an arrow */
    noArrow: PropTypes.bool,
    /** The options are 'auto',
    'auto-start',
    'auto-end',
    'bottom-start',
    'bottom',
    'bottom-end',
    'left-start',
    'left',
    'left-end',
    'right-start',
    'right',
    'right-end',
    'top-start',
    'top',
    'top-end'
    You can also pass an array of these to specify which placements to fallback to
    */
    placement: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.oneOf(POPPER_PLACEMENTS)),
        PropTypes.oneOf(POPPER_PLACEMENTS)
    ]),
    /** Additional classeNames to be spread to the overlay element */
    popperClassName: PropTypes.string,
    /** Additional props to be spread to the overlay element, supported by <a href="https://popper.js.org" target="_blank">popper.js</a> */
    popperProps: PropTypes.object,
    /** Handling for show/hide popover if true show the popover */
    show: PropTypes.bool,
    /**  Indicates the type of popup - "dialog", "grid", "listbox", "menu", or "tree".
     * This value is attached to aria-haspopup and is useful to assistive tech. Defaulted to boolean true*/
    type: PropTypes.oneOf(POPOVER_TYPES),
    useArrowKeyNavigation: PropTypes.bool,
    /** 'none', 'matchTarget', 'minTarget', 'maxTarget'
     * - "matchTarget" - left and right edges align with the target
     * - "minTarget" - right edge aligns with target unless Popover content is bigger
     * - "maxTarget" - right edge aligns with target unless Popover content is smaller
     */
    widthSizingType: PropTypes.oneOf(POPPER_SIZING_TYPES),
    /**
     * Callback function; triggered on clicking outside of popover body.
     *
     * @param {MouseEvent} event
     * @returns {void}
     * */
    onClickOutside: PropTypes.func,
    /**
     * Callback function; triggered when `escape` key is pressed and popover body is visible.
     *
     * @returns {void}
     * */
    onEscapeKey: PropTypes.func
};

Popover.defaultProps = {
    widthSizingType: 'none',
    popperProps: {},
    onClickOutside: () => { },
    onEscapeKey: () => { }
};

export default withStyles(Popover);
