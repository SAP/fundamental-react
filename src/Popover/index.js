export { default as Popover } from './Popover.js';
