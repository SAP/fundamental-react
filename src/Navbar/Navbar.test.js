import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Navbar, NavbarGroup, NavbarActions, NavbarElement } from './Navbar';
import { Button } from '../Button/Button';
import { Popover } from '../Popover/Popover';
import { Menu, MenuList, MenuItem } from '../Menu/Menu';
import { Identifier } from '../Identifier/Identifier';

Enzyme.configure({ adapter: new Adapter() });

describe('<Navbar />', () => {
  const defaultNavBar = (
    <Navbar>
      <NavbarGroup alignment="left">
        <NavbarElement type="side-menu">
          <Button option="light" glyph="menu2" navbar />
        </NavbarElement>
        <NavbarElement type="logo" noMargin="left" />
        <NavbarElement type="product-name">Product Name</NavbarElement>
      </NavbarGroup>

      <NavbarGroup launchpad>
        <Popover
          control={<Button option="light">Suite Name</Button>}
          noArrow
          body={
            <Menu>
              <MenuList>
                <MenuItem url="/">Option 1</MenuItem>
                <MenuItem url="/">Option 2</MenuItem>
                <MenuItem url="/">Option 3</MenuItem>
                <MenuItem url="/">Option 4</MenuItem>
              </MenuList>
            </Menu>
          }
        />
      </NavbarGroup>

      <NavbarGroup alignment="right">
        <NavbarElement type="context-menu">
          <Popover
            control={<Button option="light">Context Switcher</Button>}
            body={
              <Menu>
                <MenuList>
                  <MenuItem url="/">Option 1</MenuItem>
                  <MenuItem url="/">Option 2</MenuItem>
                  <MenuItem url="/">Option 3</MenuItem>
                  <MenuItem url="/">Option 4</MenuItem>
                </MenuList>
              </Menu>
            }
          />
        </NavbarElement>

        <NavbarActions>
          <Button option="light" glyph="search" navbar />
          <Button option="light" glyph="action-settings" navbar />
          <Button option="light" navbar>
            <Identifier size="s" modifier="circle">
              WW
            </Identifier>
          </Button>
        </NavbarActions>
      </NavbarGroup>
    </Navbar>
  );
  test('create navbar', () => {
    const component = renderer.create(defaultNavBar);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
