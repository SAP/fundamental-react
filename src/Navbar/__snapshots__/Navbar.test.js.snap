// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> create navbar 1`] = `
<nav
  className="fd-global-nav"
>
  <div
    className="fd-global-nav__group fd-global-nav__group--left"
  >
    <div
      className="fd-global-nav__side-menu"
    >
      <button
        className="fd-button--light sap-icon--menu2 fd-global-nav__btn"
        disabled={false}
        selected={false}
      />
    </div>
    <div
      className="fd-global-nav__logo fd-has-margin-left-none"
    />
    <div
      className="fd-global-nav__product-name"
    >
      Product Name
    </div>
  </div>
  <div
    className="fd-global-nav__group fd-global-nav__launchpad"
  >
    <div
      className="fd-popover"
    >
      <div
        aria-expanded={false}
        className="fd-popover__control"
        onClick={[Function]}
      >
        <button
          className="fd-button--light"
          disabled={false}
          selected={false}
        >
          Suite Name
        </button>
      </div>
      <div
        aria-hidden={true}
        className="fd-popover__body fd-popover__body--no-arrow"
      >
        <nav
          className="fd-menu"
        >
          <ul
            className="fd-menu__list"
          >
            <li>
              <a
                className="fd-menu__item"
                href="/"
              >
                Option 1
              </a>
            </li>
            <li>
              <a
                className="fd-menu__item"
                href="/"
              >
                Option 2
              </a>
            </li>
            <li>
              <a
                className="fd-menu__item"
                href="/"
              >
                Option 3
              </a>
            </li>
            <li>
              <a
                className="fd-menu__item"
                href="/"
              >
                Option 4
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </div>
  <div
    className="fd-global-nav__group fd-global-nav__group--right"
  >
    <div
      className="fd-global-nav__context-menu"
    >
      <div
        className="fd-popover"
      >
        <div
          aria-expanded={false}
          className="fd-popover__control"
          onClick={[Function]}
        >
          <button
            className="fd-button--light"
            disabled={false}
            selected={false}
          >
            Context Switcher
          </button>
        </div>
        <div
          aria-hidden={true}
          className="fd-popover__body"
        >
          <nav
            className="fd-menu"
          >
            <ul
              className="fd-menu__list"
            >
              <li>
                <a
                  className="fd-menu__item"
                  href="/"
                >
                  Option 1
                </a>
              </li>
              <li>
                <a
                  className="fd-menu__item"
                  href="/"
                >
                  Option 2
                </a>
              </li>
              <li>
                <a
                  className="fd-menu__item"
                  href="/"
                >
                  Option 3
                </a>
              </li>
              <li>
                <a
                  className="fd-menu__item"
                  href="/"
                >
                  Option 4
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </div>
    <div
      className="fd-global-nav__actions"
    >
      <button
        className="fd-button--light sap-icon--search fd-global-nav__btn"
        disabled={false}
        selected={false}
      />
      <button
        className="fd-button--light sap-icon--action-settings fd-global-nav__btn"
        disabled={false}
        selected={false}
      />
      <button
        className="fd-button--light fd-global-nav__btn"
        disabled={false}
        selected={false}
      >
        <span
          className="fd-identifier--s fd-identifier--circle"
          role=""
        >
          WW
        </span>
      </button>
    </div>
  </div>
</nav>
`;
